plugins/providers/bigquery/client.go

```diff
@@ -327,6 +306,34 @@ func (c *bigQueryClient) ListAccess(ctx context.Context, resources []*domain.Res
 	return access, nil
 }
 
+func (c *bigQueryClient) getGrantAbleRoleForDataset() ([]bq.AccessRole, error) {
+	var roles = []bq.AccessRole{bq.OwnerRole, bq.WriterRole, bq.ReaderRole}
+	var resourceName string
+	ctx := context.Background()
```

Comment:
why create a new `context` instead of propagating it?

Reply:
@mabdh there is no `context` from the caller of this method

`(p *Provider) CreateConfig(pc *domain.ProviderConfig) error`

To avoid change the signature of it creating a context in method itself

Reply:
fixed